crates/sdk-core-c-bridge/src/client.rs

```diff
@@ -249,6 +252,18 @@ pub extern "C" fn temporal_core_client_update_metadata(
         .set_headers(metadata.to_string_map_on_newlines());
 }
 
+#[unsafe(no_mangle)]
+pub extern "C" fn temporal_core_client_update_binary_metadata(
```

Comment:
I think, like github.com/temporalio/sdk-ruby/pull/348 and others, we should just differentiate plain vs binary metadata based on whether the key suffixed with `-bin`. This is what the gRPC spec does and most gRPC clients do. From a user POV, they only need to provide one metadata map.